### ISSUE TITLE
Automatically build getting-started configuration package in CI

### ISSUE THREAD

**Issue by hasheddan** [upbound/universal-crossplane#102]
### What problem are you facing?

Currently the manifests for the `getting-started` package reside in the `docs/` directory. However, they are not automatically built and pushed when changes are applied, meaning the manifests could get out of sync with the latest pushed version.

### How could Universal Crossplane help solve your problem?

We should build and push this package as part of CI. It may make sense to couple the versioning to that of UXP, since it will always be used in the context of UXP. This is similar to what we do with Crossplane getting started [configurations](https://github.com/crossplane/crossplane/blob/master/.github/workflows/configurations.yml).

### Manual building workaround
Using steps from https://cloud.upbound.io/docs/uxp/build-configuration/#build-the-configuration and
https://cloud.upbound.io/docs/upbound-registry/publishing-a-listing/#push-to-your-repository, we can do something like this to manually build the package as a workaround in the meantime:

```
UBC_EMAIL=jane@example.org
docker login registry.upbound.io -u ${UBC_EMAIL}

up xpkg build --package-root="./docs/getting-started/configuration"  --name getting-started.xpkg

UBC_ACCOUNT=exampleorg
UBC_REPO=examplerepo
VERSION_TAG=v0.0.1
up xpkg push ${UBC_ACCOUNT}/${UBC_REPO}:${VERSION_TAG} -f ./docs/getting-started/configuration/getting-started.xpkg
```

**Comment by negz** [comment]
I'd like to do this, though we might want to do it from the repo in which the Upbound docs are defined (which we are considering open sourcing, but have not yet).

**Comment by jbw976** [comment]
How were the current packages for registry.upbound.io/upbound/getting-started:latest built? Were they just done manually with `kubectl crossplane` commands? This is relevant to testing and publishing the fix for #220

**Comment by elohmrow** [comment]
@jbw976 i was able to build this like so:
```ghost@theMachine universal-crossplane % k crossplane build configuration --package-root="./docs/getting-started/configuration"```

and then push it to an org's registry:
```ghost@theMachine configuration % up xpkg push registry.upbound.io/flakkari-org/getting-started-with-aws:${VERSION_TAG}```

<img width="1438" alt="Screenshot 2022-02-06 at 08 19 57" src="https://user-images.githubusercontent.com/1139586/152682907-b5bdce01-3c0c-4a23-9ab8-8bf7b5defd69.png">

note that i was _not_ able to build the same way, as say `platform-ref-azure` (using `up xpkg build`). for that, i cloned a standalone repo, made changes, etc. this pkg isn't a standalone repo, so i got funny errors like:
```
ghost@theMachine universal-crossplane % up xpkg build --name getting-started.xpkg --ignore ".github/workflows/*,examples/*,hack/*" 
up: error: failed to build package: failed to parse package: {path:/Users/ghost/upbound/universal-crossplane/.golangci.yml position:0}: couldn't get version/kind; json parse error: json: cannot unmarshal array into Go value of type struct { APIVersion string "json:\"apiVersion,omitempty\""; Kind string "json:\"kind,omitempty\"" }
```


note: the screenshot has `platform-ref-aws` and not `platform-ref-azure` because i'm using 2 diff orgs.

**Comment by jbw976** [comment]
That looks like a perfectly cromulent way to build, push, and test this package @elohmrow, thanks for providing that info!

as per #220, when you installed this package, were you also able to confirm that the version of provider-aws that is automatically installed via the dependency is a newer version beyond v0.18.1?

**Comment by elohmrow** [comment]
> as per https://github.com/upbound/universal-crossplane/issues/220, when you installed this package, were you also able to confirm that the version of provider-aws that is automatically installed via the dependency is a newer version beyond v0.18.1?

nope!

it just hangs - last i checked, for 12 minutes:
```
ghost@theMachine universal-crossplane % k get pkg
NAME                                              INSTALLED   HEALTHY   PACKAGE                                                            AGE
configuration.pkg.crossplane.io/getting-started                         registry.upbound.io/flakkari-org/getting-started-with-aws:latest   12m
ghost@theMachine universal-crossplane %
```

<img width="1440" alt="Screenshot 2022-02-06 at 16 10 13" src="https://user-images.githubusercontent.com/1139586/152701530-57364558-1825-4129-afb6-9f756a9afb77.png">

fwiw, the entire diff is:
```
ghost@theMachine universal-crossplane % git diff
diff --git a/docs/getting-started/configuration/crossplane.yaml b/docs/getting-started/configuration/crossplane.yaml
index e29633d..640518a 100644
--- a/docs/getting-started/configuration/crossplane.yaml
+++ b/docs/getting-started/configuration/crossplane.yaml
@@ -9,4 +9,4 @@ spec:
     version: ">=v1.2.1-0"
   dependsOn:
     - provider: registry.upbound.io/crossplane/provider-aws
-      version: "v0.18.1"
\ No newline at end of file
+      version: ">=v0.22.0-0"
diff --git a/docs/getting-started/install.yaml b/docs/getting-started/install.yaml
index 65842e3..2b696a9 100644
--- a/docs/getting-started/install.yaml
+++ b/docs/getting-started/install.yaml
@@ -3,4 +3,5 @@ kind: Configuration
 metadata:
   name: getting-started
 spec:
-  package: registry.upbound.io/upbound/getting-started:latest
+  #package: registry.upbound.io/flakkari-org/getting-started-with-aws:latest
+  package: registry.upbound.io/flakkari-org/getting-started-with-aws:v0.0.4
ghost@theMachine universal-crossplane %
```

😭

**Comment by jbw976** [comment]
Let's move this discussion over to #220 because it's more relevant there - this issue here is about setting up the CI/CD builds, but this discussion is about testing the fix for #220.  See you over at #220 @elohmrow!

**Comment by jeanduplessis** [comment]
Closing as no longer relevant
